debezium-core/src/main/java/io/debezium/connector/LegacyV1AbstractSourceInfoStructMaker.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector;
+
+import java.util.Objects;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+/**
+ * Legacy source info that does not enforce presence of the version and connector fields
+ *
+ * @author Jiri Pechanec
+ */
+public abstract class LegacyV1AbstractSourceInfoStructMaker<T extends AbstractSourceInfo> implements SourceInfoStructMaker<T> {
```

Comment:
Why "LegacyV1..."? Is there a non-legacy v1? Or legacy v2? I think just "Legacy..." is fine.

Reply:
@gunnarmorling I tghout that over the time we might introduce more changes and keep them in it. But if decide there is always current and legacy and whenver current changes the previous current becomes legacy then I am fin with that. That's also the reason why I opted for class name instaed of enum.

Reply:
That's an interesting point; I still don't think we need or should expose class names directly, but perhaps the enum names we use should leave the door open for future versions. "v1" (for the currently, soon legacy one) and "v2" for the new one then perhaps? That being said, my hope is there won't be need for a v3 actually, as we should hopefully be able to refrain from further re-ordering and additions of optional new fields should be fine.

Reply:
@gunnarmorling Last question, are we going to use version field of the schema for the source?

Reply:
I haven't heard back from Randall on this yet, so let's leave it as is. After all, we also didn't increase it for earlier changes.